src/dos/drive_local.cpp

```diff
@@ -572,14 +550,14 @@ bool localDrive::AllocationInfo(uint16_t* _bytes_sector, uint8_t* _sectors_clust
 
 bool localDrive::FileExists(const char* name)
 {
-	char newname[CROSS_LEN];
-	safe_strcpy(newname, basedir);
-	safe_strcat(newname, name);
-	CROSS_FILENAME(newname);
-	dirCache.ExpandNameAndNormaliseCase(newname);
+	const std::string system_filename = GetSystemFilename(name);
```

Comment:
I was wondering about this `GetSystemFilename` function myself. Does it map the passed-in name to the name on the host filesystem? If so, could we rename it to `MapDosToHostFilename()`? 

It would be nice to use "DOS filename" and "host filename" consistently and "system file" has unfortunate connotations and is rather unclear. First I thought this function deals with some special system files and it confused the hell out of me 😄

Reply:
That's a good point.  Renamed as you suggested and also renamed `GetSystemFilePtr` to `GetHostFilePtr`